### DISCUSSION TITLE
Getting build error when adding jsonforms-property-view dependency to Theia GLSP client

### DISCUSSION THREAD

**Discussion by Ginxss** [eclipse-glsp/glsp#869] [Q&A] [answered]
Hi,

I want to use the [jsonforms-property-view](https://github.com/eclipse-emfcloud/jsonforms-property-view) in a Theia GLSP Client.

The error I am experiencing is very easy to reproduce:
Simply take the current [java-emf-theia project template](https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/java-emf-theia), add `"@eclipse-emfcloud/jsonforms-property-view": "0.7.0"` to the dependencies of the tasklist-theia module and build.

You will see the following error:
```
ERROR in ../node_modules/json-schema-ref-parser/lib/resolvers/http.js
Module not found: Error: Can't resolve 'http' in '[path-to-project]\java-emf-theia\glsp-client\node_modules\json-schema-ref-parser\lib\resolvers'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "https": require.resolve("https-browserify") }'
        - install 'https-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "https": false }
```
The same is output for 'https' and 'url'.

Going into the webpack.config.js and adding either
```
config.resolve.fallback.http = require.resolve('stream-http');
config.resolve.fallback.https = require.resolve('https-browserify');
config.resolve.fallback.url = require.resolve('url/');
```
or 
```
config.resolve.fallback.http = false;
config.resolve.fallback.https = false;
config.resolve.fallback.url = false;
```
as suggested by the error message does fix the build issue, but it breaks something else as the Theia browser app is just infinitely loading in both cases.

I can build the jsonforms-property-view example project without problems. Interestingly, in the node_modules of this example project there is no "json-schema-ref-parser" module at all, but in the GLSP project template, there is, even though I request the same version of the jsonforms property view.

How do I add the jsonforms property view dependency to the java-emf-theia project template without it breaking?

Thanks for your help!

**Comment by ndoschek** [comment, marked as answer]
Hi @Ginxss,

Unfortunately `0.7.0` is a rather outdated version of the package (older than 2 years now).

I suggest to use these dependencies to overcome the issues and make it work with the mentioned project-template:

```
  "dependencies": {
   ...
    "@eclipse-emfcloud/jsonforms-property-view": "0.8.0-theia-cr01",
    "@jsonforms/core": "^3.0.0",
    "@jsonforms/react": "^3.0.0",
    "@jsonforms/vanilla-renderers": "^3.0.0",
    ...
  },
```

As reference, the latest next version is currently `0.8.0-next.2bc947c.36`. See also the [list of available npm versions](https://www.npmjs.com/package/@eclipse-emfcloud/jsonforms-property-view?activeTab=versions).

HTH and best regards,
Nina
